codec-socks/src/main/java/io/netty/handler/codec/socks/SocksCmdRequestDecoder.java

```diff
@@ -28,66 +28,58 @@
  */
 public class SocksCmdRequestDecoder extends ReplayingDecoder<State> {
 
-    private SocksProtocolVersion version;
-    private int fieldLength;
-    private SocksCmdType cmdType;
-    private SocksAddressType addressType;
-    @SuppressWarnings("UnusedDeclaration")
-    private byte reserved;
-    private String host;
-    private int port;
-    private SocksRequest msg = SocksCommonUtils.UNKNOWN_SOCKS_REQUEST;
-
     public SocksCmdRequestDecoder() {
         super(State.CHECK_PROTOCOL_VERSION);
     }
 
     @Override
     protected void decode(ChannelHandlerContext ctx, ByteBuf byteBuf, List<Object> out) throws Exception {
+        SocksCmdType cmdType = null;
+        SocksAddressType addressType = null;
         switch (state()) {
             case CHECK_PROTOCOL_VERSION: {
-                version = SocksProtocolVersion.valueOf(byteBuf.readByte());
+                SocksProtocolVersion version = SocksProtocolVersion.valueOf(byteBuf.readByte());
                 if (version != SocksProtocolVersion.SOCKS5) {
                     break;
```

Comment:
do we need a `out.add(SocksCommonUtils.UNKNOWN_SOCKS_REQUEST);` here to preserve behavior?